Code/Java/Cycle_Detection_In_Linked_List.java

```diff
@@ -0,0 +1,131 @@
+/*    Floyd’s Cycle-Finding Algorithm 
+      Approach: This is the fastest method and optimized approach to traverse a LinkedList:  
+        * Traverse linked list using two pointers.
+        * Move one pointer(slow_p) by one and another pointer(fast_p) by two.
+        * If these pointers meet at the same node then there is a loop. If pointers do not meet then linked list doesn’t have a loop.   
+*/
+import java.io.*;
+import java.util.*;
+public class Cycle_Detection_In_Linked_List {
+    // Creation of SinglyLinkedListNode 
+    static class SinglyLinkedListNode {
+        public int data;
+        public SinglyLinkedListNode next;
+        
+        public SinglyLinkedListNode(int nodeData) {
+            this.data = nodeData;
+            this.next = null;
+        }
+    }
+    // Creation of Singly Linked List with insertNode function  
+    static class SinglyLinkedList {
+        public SinglyLinkedListNode head;
+        public SinglyLinkedListNode tail;
+
+        public SinglyLinkedList() {
+            this.head = null;
+            this.tail = null;
+        }
+         
+        public void insertNode(int nodeData) {
+            SinglyLinkedListNode node = new SinglyLinkedListNode(nodeData);
+
+            if (this.head == null) {
+                this.head = node;
+            } else {
+                this.tail.next = node;
+            }
+            this.tail = node;
+        }
+    }
+    static boolean hasCycle(SinglyLinkedListNode head) {
+            if(head == null ){
+                 return false;
+              }
+              SinglyLinkedListNode slow = head;
+              SinglyLinkedListNode fast = head.next;
+              while(slow != fast) {
+                    if(fast == null || fast.next == null) {
```

Comment:
check this condition once.

Reply:
I have updated the condition, please review it.